src/screens/Inventory/AddInventory.tsx

```diff
@@ -0,0 +1,99 @@
+import { createStyles, FormControl, TextField as MaterialTextField, Theme } from '@material-ui/core';
+import { withStyles } from '@material-ui/core/styles';
+import { Autocomplete } from '@material-ui/lab';
+import React, { useState } from 'react';
+import { connect } from 'react-redux';
+import { RouteComponentProps, useHistory } from 'react-router-dom';
+import BaseScreen from '../../components/BaseComponents/BaseScreen';
+import Button from '../../components/Button';
+import TextField from '../../components/TextField';
+import { InventoryRecord, ProductRecord, SiteRecord } from '../../lib/airtable/interface';
+import { createInventory } from '../../lib/airtable/request';
+import { addInventoryToRedux } from '../../lib/redux/inventoryData';
+import { EMPTY_INVENTORY, SiteInventoryData } from '../../lib/redux/inventoryDataSlice';
+import { EMPTY_SITE } from '../../lib/redux/siteDataSlice';
+import { RootState } from '../../lib/redux/store';
+
+const styles = (theme: Theme) =>
+  createStyles({
+    content: {
+      color: theme.palette.text.primary,
+    },
+    formControl: {
+      width: '100%',
+    },
+  });
+
+interface AddInventoryProps extends RouteComponentProps {
+  classes: { content: string, formControl: string};
+  location: any;
+  currentSite: SiteRecord;
+  products: Record<string, ProductRecord>;
+  userId: string;
+  sitesInventory: Record<string, SiteInventoryData>;
+}
+
+
+function AddInventory(props: AddInventoryProps) {
+  const { classes, currentSite, products, userId, sitesInventory } = props;
+  const history = useHistory();
+  // Products that the site already has inventory for
+  const currentSiteProducts = sitesInventory[currentSite.id].siteInventory.map((inventory: InventoryRecord) => inventory.productId);
+  {/* Only show products that the site doesn't already have */}
+  const productOptions = Object.entries(products).filter(([id, _]) => !currentSiteProducts.includes(id)).map(item => item[0]);
+  const [selectedProductId, setSelectedProductId] = useState("");
+  const [startingAmount, setStartingAmount] = useState(0);
+
+  // TODO: Add form input validation and error messaging
+  const handleSubmit = (event: React.MouseEvent) => {
+    // Prevent page refresh on submit
+    event.preventDefault();
+
+    // Make a deep copy of an empty inventory record
+    const inventory = JSON.parse(JSON.stringify(EMPTY_INVENTORY));
+    inventory.productId = selectedProductId;
+    inventory.siteId = currentSite.id;
+    inventory.currentQuantity = startingAmount;
+    inventory.periodStartQuantity = startingAmount;
+    addInventoryToRedux(inventory);
+    createInventory(inventory);
+
+    // TODO create inventory update
+    history.replace(`item`, { inventoryItem: inventory });
```

Comment:
I know you probably took this idea from the `customerProfile`, but we're going to move to having a `currentCustomer` in redux and displaying the `currentCustomer` instead of passing state. We're doing this because the `Edit Customer` workflow is reached via the `BaseHeader` component, and there's not a good way to pass `customer` state to `BaseHeader`. 

The inventory should follow the same paradigm so everything is consistent. You can make a `setCurrentInventoryItem` reducer to set the inventory item appropriate. You should also make sure that when a `Site` context is changed via `setCurrSite`, we should null out the `currentInventoryItem`. This can be done via passing an `extraReducer` to `siteDataSlice` according to [these docs](https://redux-toolkit.js.org/api/createSlice#the-extrareducers-map-object-notation). I'll also have a draft PR posted of this for `customerDataSlice` by 3/22 if you want to wait for that

Reply:
Tabling this change to address in a dedicated PR for inventory profile 👍